adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java

```diff
@@ -435,8 +465,11 @@ private void handleConnectionRequestResult(final MqttEndpoint endpoint,
             final Span currentSpan) {
 
         final MqttContext context = MqttContext.fromConnectPacket(endpoint);
-        context.setTracingContext(currentSpan.context());
-        final Future<DeviceUser> authAttempt = authenticate(context, currentSpan);
+        final Future<Optional<Integer>> traceSamplingPrio = applyTenantTraceSamplingPriority(context, currentSpan);
```

Comment:
Why not use `OptionalInt` instead?

Reply:
Good point. As this change also involves other classes, I would change that in a subsequent commit